features/dd-sdk-android-trace/src/test/kotlin/com/datadog/android/trace/impl/DatadogSpanAdapterTest.kt

```diff
@@ -0,0 +1,229 @@
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.doReturn
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+class DatadogSpanAdapterTest {
+
+    @Mock
+    lateinit var mockAgentSpan: AgentSpan
+
+    @Mock
+    lateinit var mockDDTraceId: DDTraceId
+
+    @Mock
+    lateinit var mockDDSpan: DDSpan
+
+    @Mock
+    lateinit var mockSpanContext: AgentSpan.Context
+
+    @IntForgery
+    private var fakeInt: Int = 0
+
+    @BoolForgery
+    private var fakeBool: Boolean = false
+
+    @StringForgery
+    lateinit var fakeString: String
+
+    @LongForgery
+    private var fakeLong: Long = 0L
+
+    @Forgery
+    lateinit var fakeThrowable: Throwable
+
+    private lateinit var testedSpanAdapter: DatadogSpanAdapter
+    private lateinit var testedSpanAdapterWithDDSpanDelegate: DatadogSpanAdapter
+
+    @BeforeEach
+    fun `set up`() {
+        whenever(mockAgentSpan.context()).thenReturn(mockSpanContext)
+        whenever(mockAgentSpan.traceId).thenReturn(mockDDTraceId)
+        testedSpanAdapter = DatadogSpanAdapter(mockAgentSpan)
+        testedSpanAdapterWithDDSpanDelegate = DatadogSpanAdapter(mockDDSpan)
+    }
+
+    @Test
+    fun `M delegate drop W drop is called`() {
+        testedSpanAdapter.drop()
+
+        verify(mockAgentSpan).drop()
+    }
+
+    @Test
+    fun `M delegate finish W finish() is called`() {
+        testedSpanAdapter.finish()
+
+        verify(mockAgentSpan).finish()
+    }
+
+    @Test
+    fun `M delegate finish W finish(millis) is called`() {
+        testedSpanAdapter.finish(fakeLong)
+
+        verify(mockAgentSpan).finish(fakeLong)
+    }
+
+    @Test
+    fun `M delegate setTag(String, Number) W drop is called`() {
+        testedSpanAdapter.setTag(fakeString, fakeLong)
+
+        verify(mockAgentSpan).setTag(fakeString as? String, fakeLong as? Number)
+    }
+
+    @Test
+    fun `M delegate setTag(String, String) W drop is called`() {
+        testedSpanAdapter.setTag(fakeString, fakeString)
+
+        verify(mockAgentSpan).setTag(fakeString, fakeString)
+    }
+
+    @Test
+    fun `M delegate setTag(String, Object) W drop is called`() {
+        val value = Any()
+        testedSpanAdapter.setTag(fakeString, value)
+
+        verify(mockAgentSpan).setTag(fakeString, value)
+    }
+
+    @Test
+    fun `M delegate setMetric(String, Int) W drop is called`() {
+        testedSpanAdapter.setMetric(fakeString, fakeInt)
+
+        verify(mockAgentSpan).setMetric(fakeString as CharSequence, fakeInt)
```

Comment:
why this cast is needed? is there method overload ambiguity here?

Reply:
debug leftover, removed